lib/src/compatKtLint0Dot34Dot2/java/com/diffplug/spotless/glue/ktlint/compat/KtLintCompat0Dot34Dot2Adapter.java

```diff
@@ -55,13 +56,13 @@ public String format(final String text, final String name, final boolean isScrip
 		}
 
 		return KtLint.INSTANCE.format(new KtLint.Params(
-				name,
+				path.toFile().getAbsolutePath(),
```

Comment:
Generally, I think it's better to stay in the Java NIO domain as much as possible, and I believe it would be equivalent if you used `path.toAbsolutePath().toString()` in all places so that it doesn't go into the `File` API. What do you think?

Reply:
Why do you think this way?

KtLint 0.48+ use Path for the original file, which must have an absolute path as per EditorConfig specification and official library used.
It also uses Path interface for EditorConfig file itself.

Reply:
I prefer to use newer version of KtLint for the primary interface and work backward compatibility for the rest as possible

Reply:
It's a good idea to switch all File to Path everywhere, but I don't to refactor whole code _right now_.

Reply:
Yes, I think that's wise.

Reply:
> It's a good idea to switch all File to Path everywhere, but I don't to refactor whole code right now.

Ah, yes, I wasn't meaning that the whole codebase should be refactored. I was just referring to the adapter code where you had changes through this PR.

Reply:
As someone who contributed to Spotless in the past, my understanding of why we use `File` is because Gradle's APIs generally use `File` over `Path` for historical reasons.

That being said, this class is in spotless-lib, so theoretically there's nothing stopping us from migrating it fully to `Path` in the future. :)